test/plugins.spec.js

```diff
@@ -155,33 +172,48 @@ test('override duplicates in smartContract.addContracts with the new address', a
 
   await Maker.create('test', {
     autoAuthenticate: false,
-    plugins: [testPlugin],
+    provider: { url: 'userURL' },
+    plugins: [testPlugin1, testPlugin2],
     smartContract: {
       addContracts: {
-        // different address
-        foo: {
-          abi: [],
-          address: '0xbeefed1bedded2dabbed3defaced4decade5bead'
-        },
-        // different ABI
-        bar: {
-          abi: [{ ...exampleAbiItem, name: 'zest' }],
-          address: '0xbeefed1bedded2dabbed3defaced4decade5bade'
+        FOO: {
+          address: '0xbeefed1bedded2dabbed3defaced4decade5foo3'
         },
-        // same address and ABI -- will not cause error
-        baz: {
-          abi: [],
-          address: '0xbeefed1bedded2dabbed3defaced4decade5abed'
+        TOP: {
+          address: '0xbeefed1bedded2dabbed3defaced4decade5top3',
+          abi: [exampleUserConfigAbi]
         }
       }
     }
   });
 
   const last = ConfigFactory.create.mock.calls.length - 1;
-  expect(
-    ConfigFactory.create.mock.calls[last][1].smartContract.addContracts.foo
-      .address
-  ).toBe(newAddress);
+
+  const addContractsResult =
+    ConfigFactory.create.mock.calls[last][1].smartContract.addContracts;
+
+  // User supplied overrides all
+  const fooExpected = '0xbeefed1bedded2dabbed3defaced4decade5foo3';
+  const topExpected = '0xbeefed1bedded2dabbed3defaced4decade5top3';
+  // 2nd plugin overrides the 1st
+  const tubExpected = '0xbeefed1bedded2dabbed3defaced4decade5tub2';
+  const barExpected = '0xbeefed1bedded2dabbed3defaced4decade5bar2';
+  // No overrides for this one from 1st plugin
+  const tapExpected = '0xbeefed1bedded2dabbed3defaced4decade5tap1';
+
+  expect(addContractsResult['FOO'].address).toBe(fooExpected);
+  expect(addContractsResult['TOP'].address).toBe(topExpected);
```

Comment:
for these ones that refer to core contracts, i would get the actual value of e.g. `maker.service('smartContract').getContractAddress('TOP')` to verify that it overrode the original value